src/ncp-spinel/SpinelNCPInstance.cpp

```diff
@@ -821,7 +821,7 @@ SpinelNCPInstance::set_property(
 		} else if (strcaseequal(key.c_str(), kWPANTUNDProperty_NetworkXPANID)) {
 			Data xpanid = any_to_data(value);
 
-			mXPanid = any_to_uint64(value);
+			memcpy(mCurrentNetworkInstance.hwaddr, xpanid.data(), xpanid.size());
```

Comment:
This is wrong for two reasons:

1. You are setting the xpanid to the wrong field. `hwaddr` is only used by scan results.
2. The same reason as my comment above for the PANID: The source of truth is the value of the property on the NCP, not in wpantund. We set the XPANID on the NCP and let the NCP tell us the XPANID in the response.